src/ngraph/runtime/aligned_buffer.cpp

```diff
@@ -17,24 +17,37 @@
 #include <memory>
 
 #include "ngraph/runtime/aligned_buffer.hpp"
+#include "ngraph/runtime/allocator.hpp"
 #include "ngraph/util.hpp"
 
 using namespace ngraph;
 
 runtime::AlignedBuffer::AlignedBuffer()
     : m_allocated_buffer(nullptr)
     , m_aligned_buffer(nullptr)
+    , m_allocator(nullptr)
     , m_byte_size(0)
 {
 }
 
-runtime::AlignedBuffer::AlignedBuffer(size_t byte_size, size_t alignment)
+runtime::AlignedBuffer::AlignedBuffer(size_t byte_size,
+                                      size_t alignment,
+                                      std::shared_ptr<ngraph::runtime::Allocator> allocator)
 {
+    m_allocator = allocator;
     m_byte_size = byte_size;
     if (m_byte_size > 0)
     {
         size_t allocation_size = m_byte_size + alignment;
-        m_allocated_buffer = static_cast<char*>(ngraph_malloc(allocation_size));
+        if (m_allocator)
+        {
+            m_allocated_buffer =
+                static_cast<char*>(m_allocator->Malloc(nullptr, allocation_size, alignment));
+        }
+        else
+        {
+            m_allocated_buffer = static_cast<char*>(ngraph_malloc(allocation_size));
```

Comment:
Isn't ngraph_malloc et al. just another allocator? Would it make sense to encode it as another Allocator and assume that m_allocator will never be null?